src/backends/webgpu/src/kernels/conv2d_webgpu.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {Conv2DInfo} from '@tensorflow/tfjs-core/dist/ops/conv_util';
+import {assert} from '@tensorflow/tfjs-core/dist/util';
+
+import {WebGPUProgram} from './webgpu_program';
+
+export class Conv2DProgram implements WebGPUProgram {
+  outputShape: number[];
+  userCode: string;
+  dispatch: [number, number, number];
+  variableNames = ['inp', 'flt'];
+  uniforms = 'uvec4 inpShape, outShape; uvec2 fltShape, pad, stride;';
+  tileSize: [number, number, number] = [2, 2, 1];
+
+  constructor(convInfo: Conv2DInfo) {
+    this.outputShape = convInfo.outShape;
+    this.dispatch = [
+      Math.ceil(this.outputShape[2] / this.tileSize[0]),
+      Math.ceil(this.outputShape[1] / this.tileSize[1]),
+      Math.ceil(this.outputShape[0] * this.outputShape[3] / this.tileSize[2]),
+    ];
+
+    assert(
+        convInfo.dataFormat == 'channelsLast',
+        () => 'TODO: NCHW is unimplemented');
+    assert(
+        convInfo.dilationHeight === 1 && convInfo.dilationWidth === 1,
+        () => 'TODO: Dilation is unimplemented');
```

Comment:
Can you check this usage assert is ok before merging? In case you haven't checked it yet